waku/v2/node/peer_manager/peer_manager.nim

```diff
@@ -364,3 +390,53 @@ proc relayConnectivityLoop*(pm: PeerManager) {.async.} =
     await pm.connectToNodes(outsideBackoffPeers[0..<numPeersToConnect], WakuRelayCodec)
 
     await sleepAsync(ConnectivityLoopInterval)
+
+# Ensure we are always connected to the slotted service peers
+proc serviceConnectivityLoop*(pm: PeerManager) {.async.} =
+  pm.serviceLoopUp = true
+  info "Starting service connectivity loop"
+
+  while pm.serviceLoopUp:
+    if pm.serviceSlots.len == 0:
+      warn "No service peers configured, but service loop is running"
+    for serviceProto, servicePeer in pm.serviceSlots.pairs:
+      if pm.peerStore.connectedness(servicePeer.peerId) != Connected:
+        # Attempt to dial peer. Note that service peers do not respect any backoff
+        let conn = await pm.dialPeer(servicePeer.peerId, servicePeer.addrs, serviceProto)
+        if conn.isNone:
+          warn "Could not connect with service peer", peerId=servicePeer.peerId, proto=serviceProto
```

Comment:
this will be seen in `debug`, inside `dialPeer` function.
`debug "Dialing peer failed", peerId = peerId, reason = reasonFailed, failedAttempts=failedAttempts`